tests/reset/reset.cpp

```diff
@@ -274,6 +274,8 @@ TEST(Reset, removeResetInvalidArguments)
 {
     libcellml::ComponentPtr c = libcellml::Component::create();
     libcellml::ResetPtr r1 = libcellml::Reset::create();
+    r1->setOrder(2);
+
```

Comment:
I don't understand why this test needed to change?  If you wanted to show a different situation, why not add it?

Reply:
This test had to change to show the same behaviour.  If `r1` is not different to `r2` then `r1` is removed later on when you ask to remove a `reset` that is equal to `r2`.